sapientml_preprocess/generator.py

```diff
@@ -229,15 +230,46 @@ def generate_code(self, dataset: Dataset, task: Task) -> Tuple[Dataset, Code]:
             logger.warning(
                 f"Symbols that inhibit training and visualization will be removed from column name {str(cols_has_symbols)}."
             )
+            org_df_column = df.columns.values
+            org_target_column = task.target_columns
             df = df.rename(columns=lambda col: remove_symbols(col) if col in cols_has_symbols else col)
             task.target_columns = [
                 remove_symbols(col) if col in cols_has_symbols else col for col in task.target_columns
             ]
+            same_column = {k: v for k, v in collections.Counter(list(df.columns.values)).items() if v > 1}
+            rename_dict = {}
+            if len(same_column) != 0:
+                for target in same_column.keys():
+                    rename_dict = {}
+                    rename_target_col = []
+                    df_cols = list(df.columns.values)
+                    i = 1
+                    for col in df_cols:
+                        if target in col:
+                            rename_dict[org_df_column[len(rename_dict)]] = str(col + str(i))
+                            i = i + 1
+                        else:
+                            rename_dict[org_df_column[len(rename_dict)]] = col
+                    df = df.set_axis(list(rename_dict.values()), axis=1)
+                    i = 1
+                    for col in org_target_column:
+                        rename_target_col.append(rename_dict[col])
+
+                    task.target_columns = rename_target_col
+
```

Comment:
- I think the code may rename column names which originally don't contain symbols, right? If so, could you keep the names when the column names contain symbols.
- This code is hard to read and can have unexpected issues. Could you rewrite it?

Reply:
The review has been applied.